dragonfly_doe2/doe/mats_layers_constr.py

```diff
@@ -42,43 +44,91 @@ def lbt_to_ip(self, _val, _to, _from):
         value = base_type.to_unit(_val, _to, _from)
         return(value)
 
-    def to_inp(self):
+    def to_inp(self) -> str:
+        # TODO: for sake of OCD: revert to 3.x f'strings'
         if self.mat_type == 'PROPERTIES':
-            return'"{display_name}" = MATERIAL\n'.format(
+
+            return'\n"{display_name}" = MATERIAL\n'.format(
                 display_name=self.hbmat.display_name) + \
                 '   TYPE            = {mat_type}\n'.format(mat_type=self.mat_type) + \
                 '   THICKNESS       = {thickness}\n'.format(
-                    thickness=self.lbt_to_ip(self.hbmat.thickness, 'ft', 'm')) + \
+                    thickness=self.lbt_to_ip([self.hbmat.thickness], 'ft', 'm')) + \
                 '   CONDUCTIVITY    = {cond}\n'.format(
-                    cond=self.lbt_to_ip(self.hbmat.conductivity, 'Btu/h-ft2', 'W/m2')) + \
+                    cond=self.lbt_to_ip([self.hbmat.conductivity], 'Btu/h-ft2', 'W/m2')) + \
                 '   DENSITY         = {dens}\n'.format(dens=self.hbmat.density/16.018) + \
                 '   SPECIFIC-HEAT   = {spech}\n'.format(
-                    spech=self.lbt_to_ip(self.hbmat.specific_heat, 'Btu/lb', 'J/kg')) + \
+                    spech=self.lbt_to_ip([self.hbmat.specific_heat], 'Btu/lb', 'J/kg')) + \
                 '   ..'
         elif self.mat_type == 'RESISTANCE':
             return '"{display_name}" = MATERIAL\n'.format(
                 display_name=self.hbmat.display_name) + \
                 '   TYPE           = {mat_type}\n'.format(mat_type=self.mat_type) + \
-                '   RESISTANCE     = {r_val}\n   ..'.format(
-                    r_val=self.lbt_to_ip(self.hbmat.r_value, 'h-ft2-F/Btu', 'm2-K/W'))
+                '   RESISTANCE     = {r_val}\n   ..\n'.format(
+                    r_val=self.lbt_to_ip([self.hbmat.r_value], 'h-ft2-F/Btu', 'm2-K/W'))
 
     def __repr__(self):
         return self.to_inp()
 
 
-class Construction(object):
+@dataclass()
+class MatsLayersConstructions(object):
     """Construction object. Contains, materials and layers for *.inp file.
     Intent: Pass list of constrs in dfm to this class.
     Should return enum of mats, 'layers' (stupid f'n eQuest obj imo..), Constr objs
-    via __repr__
-    TODO: make it happen! easy day! tomorrow.
+    via __repr__.
+
+        Returns:
+          $  Materials / Layers / Constructions *.inp block
     """
+    inp_constructions: list = None
+    inp_layers: list = None
+    inp_materials: list = None
```

Comment:
We already know this is an inp object. We should keep the names clean.

```suggestion
    constructions: list = None
    layers: list = None
    materials: list = None
```

Also aren't these inputs required? Why are you putting default None values for them?

Reply:
I wasn't quite sure, tbh:
in the 'simple.dfjson'; there are no constructions/mats etc; so the intent was to basically have a 'None' return for this block; so would still get the '$  mats/layers/constrs' inp block sans the actual inputs if desired by user.
(if curious: JSON editor extension nickdemayo.vscode-json-editor top 3 rando extensions imo; up there with the pdf viewer)
![image](https://user-images.githubusercontent.com/63176386/147394855-41c2dcbb-f5cb-47c4-831b-4a3eb96f2c40.png)